File: core/src/test/java/no/elg/infiniteBootleg/TestGraphic.java
package no.elg.infiniteBootleg;

import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;

import de.tomgrill.gdxtesting.GdxTestRunner;
import no.elg.infiniteBootleg.world.ClientWorld;
import no.elg.infiniteBootleg.world.SinglePlayerWorld;
import no.elg.infiniteBootleg.world.generator.EmptyChunkGenerator;
import org.jetbrains.annotations.NotNull;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;
import org.junit.jupiter.api.extension.ExtendWith;

@ExtendWith(GdxTestRunner.class)
@TestInstance(PER_CLASS)
public class TestGraphic {

  static {
    Settings.schedulerThreads = 0;
    Settings.client = false;
    Settings.loadWorldFromDisk = false;
    Settings.debug = true;
    new ClientMain(true);
  }

  public ClientWorld world;

  @NotNull
  public ClientWorld createNewWorld() {
    world = new SinglePlayerWorld(new EmptyChunkGenerator(), 0, "World");
    return world;
  }

  @Test
  public void dummy() {
    // This dummy test is needed for @RunWith(GdxTestRunner.class) to work
  }
}

File: core/src/test/java/no/elg/infiniteBootleg/TestGraphic.kt
package no.elg.infiniteBootleg

import de.tomgrill.gdxtesting.GdxTestRunner
import no.elg.infiniteBootleg.core.Settings
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestInstance
import org.junit.jupiter.api.TestInstance.Lifecycle
import org.junit.jupiter.api.extension.ExtendWith

@ExtendWith(GdxTestRunner::class)
@TestInstance(Lifecycle.PER_CLASS)
open class TestGraphic {

  @Test
  fun dummy() {
    // This dummy test is needed for @RunWith(GdxTestRunner.class) to work
  }

  companion object {
    init {
      Settings.client = false
      Settings.loadWorldFromDisk = false
      Settings.debug = true
    }
  }
}
